include/libtorrent/piece_picker.hpp

```diff
@@ -188,9 +188,14 @@ namespace libtorrent {
 			// the number of blocks in the requested state
 			std::uint16_t requested:15;
 
-			// set to true while there is an outstanding
-			// hash check for this piece
-			std::uint16_t outstanding_hash_check:1;
+			// available for future use
+			std::uint16_t unused:1;
+
+			// number of outstanding hash jobs for this piece
+			std::uint16_t hashing:15;
```

Comment:
is there ever more than one outstanding hash job for the same piece at the same time?

Reply:
I suppose the user can call `add_piece()` at any time, to trigger overlapping calls to `torrent::verify_piece()`.
However, would it be unreasonable to, at the top of that function, check for `is_hashing()` and just exit if it's already has an outstanding hash job?

The reason I ask is because it would be really nice to save 16 bits here, and fit all the state in perfectly. This is one of those data structures that's instantiated *a lot* of times.

Reply:
With the current state of the code there can only be one hash job per piece. This counter is to handle the future case where we support block level deadlines. That code will use `async_hash2` to compute an individual block hash and there may be multiple such jobs active at a time.

Reply:
would you mind if I propose a change to use a single bit for now, and wrap the counter code in `#if 0`, like the other logic for block level requests?